web/src/components/workflow-editor/workflow-editor.tsx

```diff
@@ -56,6 +60,15 @@ export default function WorkflowEditor({
 		}
 	}, [editorActions, setEditorActions, editActionsError]);
 
+	const { data: encryptedSecrets } = useListSecretsApi();
+
+	useEffect(() => {
+		if (encryptedSecrets) {
+			setSecrets(encryptedSecrets);
+			return () => clear();
```

Comment:
Maybe we should rename `clear` from the secrets store to `clearSecretsStore` to be consistent with the workflow store naming. Also, makes it more clear here what's happening.